src/typeChecker/typeErrorChecker.ts

```diff
@@ -1182,53 +1374,46 @@ function lookupTypeAndRemoveForAllAndPredicateTypes(name: string): Type | undefi
  * For forall types, the given type arguments are substituted into the poly type,
  * and the resulting type is returned.
  */
-function lookupTypeAliasAndRemoveForAllAndPredicateTypes(
-  typeNode: tsEs.TSTypeReference,
-  name: string,
-  context: Context
-): Type {
-  const type = lookupTypeAlias(name, env)
-  if (!type) {
-    context.errors.push(new TypeNotFoundError(typeNode, name))
+function lookupTypeAliasAndRemoveForAllTypes(node: tsEs.Node, varType: Variable): Type {
+  // Check if type alias exists
+  const aliasType = lookupTypeAlias(varType.name, env)
+  if (!aliasType) {
+    context.errors.push(new TypeNotFoundError(node, varType.name))
     return tAny
   }
-  if (type.kind === 'predicate') {
-    throw new TypecheckError(typeNode, 'Type alias should not be predicate type')
-  }
-  if (type.kind === 'forall') {
-    if (!type.typeParams) {
-      throw new TypecheckError(typeNode, 'Generic type aliases must have type parameters')
-    }
-    if (
-      !typeNode.typeParameters ||
-      typeNode.typeParameters.params.length !== type.typeParams.length
-    ) {
-      context.errors.push(
-        new InvalidNumberOfTypeArgumentsForGenericTypeError(typeNode, name, type.typeParams.length)
-      )
+  // If type saved in type environment is not generic,
+  // given variable type should not have type arguments
+  if (aliasType.kind !== 'forall') {
+    if (varType.typeArgs && varType.typeArgs.length > 0) {
```

Comment:
Would it be possible to make the check more specific, instead of relying on falsey values?